packages/compass-indexes/src/components/create-search-index-modal/index.tsx

```diff
@@ -0,0 +1,186 @@
+  const [indexDefinition, setIndexDefinition] = useState<string>(
+    DEFAULT_INDEX_DEFINITION
+  );
+  const [parsingError, setParsingError] = useState<string | undefined>(
+    undefined
+  );
+
+  const onSearchIndexDefinitionChanged = useCallback(
+    (newDefinition: string) => {
+      setParsingError(undefined);
+
+      try {
+        parseShellBSON(newDefinition);
+        setIndexDefinition(newDefinition);
+      } catch (ex) {
+        setParsingError((ex as Error).message);
+      }
+    },
+    [setIndexDefinition, setParsingError]
+  );
+
+  const onSaveIndex = useCallback(() => {
+    if (parsingError) {
+      return;
+    }
+
+    const indexDefinitionDoc = parseShellBSON(indexDefinition);
+    saveIndex(indexName, indexDefinitionDoc);
+  }, [saveIndex, parsingError, indexName, indexDefinition]);
+
+  useTrackOnChange(
+    'COMPASS-SEARCH-INDEXES-UI',
+    (track) => {
+      if (isModalOpen) {
+        track('Screen', { name: 'create_search_index_modal' });
+      }
+    },
+    [isModalOpen],
+    undefined,
+    React
+  );
+
+  return (
+    <Modal
+      open={isModalOpen}
+      setOpen={closeModal}
+      data-testid="create-search-index-modal"
+    >
+      <ModalHeader
+        title="Create Search Index"
+        subtitle="Give your search index a name for easy reference"
+      />
+      <ModalBody>
+        <div className={bodyGapStyles}>
+          <Label htmlFor="name-of-search-index">Name of Search Index</Label>
+          <TextInput
+            id="name-of-search-index"
+            data-testid="name-of-search-index"
+            aria-labelledby="Name of Search Index"
+            type="text"
+            state={indexName === '' ? 'error' : 'none'}
+            errorMessage={
+              indexName === '' ? 'Please enter the name of the index.' : ''
+            }
+            value={indexName}
+            onChange={(evt: React.ChangeEvent<HTMLInputElement>) =>
+              setIndexName(evt.target.value)
+            }
+          />
+        </div>
+        <HorizontalRule className={bodyGapStyles} />
+        <div className={bodyGapStyles}>
+          <Subtitle>Index Definition</Subtitle>
+          <p className={bodyGapStyles}>
+            By default, search indexes will have the following search
+            configurations. You can refine this later.
+          </p>
+          <Link href="/" target="_blank" hideExternalIcon={true}>
+            View Atlas Search tutorials{' '}
+            <Icon size="small" glyph="OpenNewTab"></Icon>
+          </Link>
+          <CodemirrorMultilineEditor
+            data-testid="definition-of-search-index"
+            text={indexDefinition}
+            onChangeText={onSearchIndexDefinitionChanged}
+            minLines={16}
+            className={bodyGapStyles}
+          />
+          {parsingError && <WarningSummary warnings={[parsingError]} />}
+          {error && <ErrorSummary errors={[error]} />}
```

Comment:
Type here is `string | string[]`, you don't need to wrap it

```suggestion
          {parsingError && <WarningSummary warnings={parsingError} />}
          {error && <ErrorSummary errors={error} />}
```